.github/workflows/CI-tag-orchestrator.yml

```diff
@@ -5,54 +5,132 @@ on:
-      DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
-      QA_SLACK_WEBHOOK_URL: ${{ secrets.QA_SLACK_WEBHOOK_URL }}
-      GH_TOKEN: ${{ secrets.GH_TOKEN }}
-
-  build-and-publish:
-    name: Build and Publish Docker Image
+      CACHING_MODE: "CACHE_RESTORE"
+  
+  build:
     runs-on: ubuntu-latest
-    needs: execute-system-tests # Wait for e2e tests to finish
+    needs: env-setup
+
+    outputs:
+      artifact_name_solo: ${{ steps.create-tar-file.outputs.tar_file_name_solo }}
+      artifact_name_relay: ${{ steps.create-tar-file.outputs.tar_file_name_relay }}
+
     steps:
-      - name: Git checkout
-        uses: actions/checkout@v4
+      - uses: actions/checkout@v4
         with:
-          ref: ${{ github.ref }}
+          ref: ${{ github.event.inputs.git_tag || github.ref }}
           fetch-depth: 0
 
-      - name: Download zkVerify Docker image artifact
-        uses: actions/download-artifact@v4
+      - name: Set up deps cache
+        run: mkdir deps
+
+      - name: Restore common cache
+        if: ${{ !env.ACT }}
+        uses: actions/cache/restore@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            deps/
+          key: common-cache
+      
+      - name: Restore build cache
+        if: ${{ !env.ACT }}
+        uses: actions/cache/restore@v3
         with:
-          name: zkverify-image
-          path: ./
+          path: |
+            target/
+          key: build-cache
       
-      - name: Build/Publish Docker image
-        env:
-          RELEASE_BRANCH: release
-          MAINTAINERS_KEYS: ${{ vars.MAINTAINERS_KEYS }}
-          DOCKER_HUB_USERNAME: ${{ secrets.DOCKER_HUB_USERNAME }}
-          DOCKER_HUB_TOKEN: ${{ secrets.DOCKER_HUB_TOKEN }}
+      - name: Verify release build
+        run: |
+          ls target
+          ls target/release
+
+      - name: Build solo and relay chain configured images
+        id: build_images
+        run: |
+          SKIP_IMAGE_VALIDATION=true
+          "${GITHUB_WORKSPACE}/docker/scripts/bootstrap.sh" --skip-parachain --skip-compile
+        shell: bash
+
+      - name: Set Docker image names as output
+        id: save-image-names
         run: |
-          source "${GITHUB_WORKSPACE}/ci/setup_env.sh"
-          "${GITHUB_WORKSPACE}/ci/docker.sh" --image-artifact zkverify-image
+          echo "docker_image_solo=horizenlabs/zkverify:latest" >> $GITHUB_OUTPUT
+          echo "docker_image_relay=horizenlabs/zkv-relay:latest" >> $GITHUB_OUTPUT
+
+      - name: Save the Docker image as a tarball
+        id: create-tar-file
+        run: |           
+          TAR_FILE_NAME_SOLO="zkverify-solo"
+          TAR_FILE_SOLO="${{github.workspace}}/${TAR_FILE_NAME_SOLO}.tar"
+
+          TAR_FILE_NAME_RELAY="zkverify-relay"
+          TAR_FILE_RELAY="${{github.workspace}}/${TAR_FILE_NAME_RELAY}.tar"
```

Comment:
I think this can be an ENV var only for a particular scope therefore you do not need to save it under GITHUB_OUTPUT since it is static and not being dynamically defined during the run: block 🤷